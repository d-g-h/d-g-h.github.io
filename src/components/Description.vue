<template>
  <li>{{ description }}</li>
</template>

<script>
export default {
  name: 'Description',
  props: {
    description: String,
  }
}
</script>
