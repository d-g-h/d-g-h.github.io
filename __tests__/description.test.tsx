/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Description from "@/components/description";

const DESCRIPTION = "Job description";

describe("Description", () => {
  it("renders Description", () => {
    render(<Description description={DESCRIPTION} />);

    const description = screen.getByText(DESCRIPTION);

    expect(description).toBeInTheDocument();
  });
  it("renders html correctly", () => {
    const { container } = render(<Description description={DESCRIPTION} />);
    expect(container).toMatchSnapshot();
  });
});
