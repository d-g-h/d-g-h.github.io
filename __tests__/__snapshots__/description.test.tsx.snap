// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description renders html correctly 1`] = `
<div>
  <li
    class="description"
    contenteditable="true"
  >
    Job description
  </li>
</div>
`;
